go/arrow/flight/flightsql/driver/driver.go

```diff
@@ -79,28 +96,37 @@ func (r *Rows) Close() error {
 // should be taken when closing Rows not to modify
 // a buffer held in dest.
 func (r *Rows) Next(dest []driver.Value) error {
-	if r.currentRecord >= len(r.records) {
-		return io.EOF
-	}
-	record := r.records[r.currentRecord]
+	if r.currentRecord == nil || int64(r.currentRow) >= r.currentRecord.NumRows() {
+		if r.streamError != nil {
+			return r.streamError
+		}
+
+		r.releaseRecord()
 
-	if int64(r.currentRow) >= record.NumRows() {
-		return ErrOutOfRange
+		// Get the next record from the channel
+		var ok bool
+		if r.currentRecord, ok = <-r.recordChan; !ok {
+			return io.EOF // Channel closed, no more records
+		}
```

Comment:
Same thing here, if we are going to have a mutex, it needs to be locked before this

Reply:
deal. mutex removed.